src/map-proposals.ts

```diff
@@ -8,14 +8,22 @@ const extendMultimapReducer = (
 	...multimap,
 });
 
-const mapProposals = (fields: ApiBaseField[], proposals: ApiProposal[]) =>
+const mapProposals = (fields: BaseField[], proposals: Proposal[]) =>
```

Comment:
This doesn't need to be part of this PR (but you might consider making this change in a separate PR before this one)

I don't think you need to do this mapping any longer since base fields are now provided as part of the application form fields directly.

Reply:
You're right that we should've documented this function and behavior better back in the day! I believe what it's doing is coercing the data structure into a format that's easier to iterate through for building the interface tables. So this is a bad function name but basically `mapApiProposalDataForInterface` would be more explicitly correct.

We can revisit the data structure to see if that's necessary any more, but the intention was that we would reduce the proposal data to only those that matched base fields we knew about, and flattened it to `key: value` as `baseFieldKey: proposalValue` (for the most recent version only), and also removed any blank values from the set.

Which is to say: the output of this process is still desirable, but maybe there's a better modern path to get there, and/or maybe it's a good time to revisit how we present this overall! (Since, for instance, we weren't yet tackling versions.)